src/sophys/common/devices/pimega.py

```diff
@@ -146,6 +154,30 @@ class Pimega(SingleTrigger, PimegaDetector):
     def __init__(self, name, prefix, **kwargs):
         super(Pimega, self).__init__(prefix, name=name, **kwargs)
 
+    def stage(self):
+        # Make sure the current acquisition status is 'Done'
+        self._acquisition_signal.set(0).wait(timeout=30.0)
+
+        self._acquisition_signal.subscribe(
+            self._acquire_setpoint_changed, EpicsSignal.SUB_SETPOINT
+        )
+
+        return super().stage()
+
+    def unstage(self):
+        super().unstage()
+
+        self._acquisition_signal.unsubscribe(self._acquire_setpoint_changed)
+
+    def _acquire_setpoint_changed(self, value, severity, **kwargs):
+        if self._status is None or self._status.done:
+            return
+
+        if value == 1 and severity >= AlarmSeverity.MAJOR:
+            exc = Exception(self.cam.backend_status_message.get())
```

Comment:
I wonder if we should actually compose both IOC and backend messages. Otherwise, a failing communication with the modules (e.g. module link goes down) won't end up in the message.